components/core/src/clp_s/search/kql/kql.cpp

```diff
@@ -69,7 +70,7 @@ class ParseTreeVisitor : public KqlBaseVisitor {
 
 public:
     static std::string unquote_string(std::string const& text) {
-        if (text.at(0) == '"') {
+        if (false == text.empty() && '"' == text.at(0)) {
```

Comment:
Do we need to do this empty check?

Reply:
Technically no because of the way our kql grammar is written and the way this is called, but I figured we should probably handle this edge case in case we ever write an incorrect caller or change the grammar.